### PR TITLE
Update /random

### PR DESCRIPTION
**Pull request by mctaylors**
Updates in /random:
- Set default minimum number to 0. https://github.com/LabsDevelopment/Octobot/pull/138/commits/00a7b5b5ad0a65407a921de51a9fc479c5450452
- Slightly redesign embed.
  - Show maximum & minimum numbers. https://github.com/LabsDevelopment/Octobot/pull/138/commits/00a7b5b5ad0a65407a921de51a9fc479c5450452
  - Recolor & display a message when user tries to use exact same number in `first` and `second` fields for some reason. https://github.com/LabsDevelopment/Octobot/pull/138/commits/00a7b5b5ad0a65407a921de51a9fc479c5450452
  - Mention user in small title. https://github.com/LabsDevelopment/Octobot/pull/138/commits/570f6bd901bf95bbde3813a6d717d75537e77187
- Automatically detect max & min numbers. https://github.com/LabsDevelopment/Octobot/pull/138/commits/b76a5cc6f5f42c7c2e20c0249d5f531c338977da — https://github.com/LabsDevelopment/Octobot/pull/138/commits/80e0b1c889889425805f67ccfd73047adff8c644
- Add `long` support. https://github.com/LabsDevelopment/Octobot/pull/138/commits/cbf263622d4526715a142fa6956f0e13db28e9ed
- Show what default number is. https://github.com/LabsDevelopment/Octobot/pull/138/commits/6a961dac3d8a736dbf754ec0df6e0c1c8cd9828c

### CODE REVIEW

File: locale/Messages.resx [hunk 1]
```diff
@@ -654,4 +654,13 @@
   <data name="RandomOutput" xml:space="preserve">
     <value>Your random number is:</value>
   </data>
+  <data name="RandomObvious" xml:space="preserve">
```
Comment (Octol1ttle) [inline locale/Messages.resx:657]: rename to `RandomMinMaxSame`

**Comment by mctaylors** [comment]
I should stop doing that.

**Comment by mctaylors** [comment]
Don't do mobile editing, kids.

File: src/Commands/ToolsCommandGroup.cs [hunk 2]
```diff
@@ -266,25 +260,29 @@ private static void AppendBanInformation(MemberData memberData, StringBuilder bu
...
-        }
+        var max = Math.Max(first, second);
+        var min = Math.Min(first, second);
 
         var i = Random.Shared.Next(min, max + 1);
 
+        var description = new StringBuilder().Append("# ").AppendLine(i.ToString())
+            .Append("- ").AppendLine(string.Format(Messages.RandomMin, Markdown.InlineCode(min.ToString())))
+            .Append("- ").AppendLine(string.Format(Messages.RandomMax, Markdown.InlineCode(max.ToString())));
+        var embedColor = ColorsList.Blue;
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:276]: newline please

File: src/Commands/ToolsCommandGroup.cs [hunk 3]
```diff
@@ -266,25 +260,30 @@ private static void AppendBanInformation(MemberData memberData, StringBuilder bu
...
-        }
+        var max = Math.Max(first, second);
+        var min = Math.Min(first, second);
 
         var i = Random.Shared.Next(min, max + 1);
 
+        var description = new StringBuilder().Append("# ").AppendLine(i.ToString())
+            .Append("- ").AppendLine(string.Format(Messages.RandomMin, Markdown.InlineCode(min.ToString())))
+            .Append("- ").AppendLine(string.Format(Messages.RandomMax, Markdown.InlineCode(max.ToString())));
+
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:276]: mobile editing moment
```suggestion

```

File: src/Commands/ToolsCommandGroup.cs [hunk 4]
```diff
@@ -266,25 +260,30 @@ private static void AppendBanInformation(MemberData memberData, StringBuilder bu
...
-        if (min > max)
-        {
-            var failedEmbed = new EmbedBuilder().WithSmallTitle(
-                    Messages.RandomMinGreaterThanMax, currentUser)
-                .WithColour(ColorsList.Red).Build();
-
-            return await _feedback.SendContextualEmbedResultAsync(failedEmbed, ct);
-        }
+        var max = Math.Max(first, second);
+        var min = Math.Min(first, second);
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:269]:
```suggestion
        var min = Math.Min(first, second);
        var max = Math.Max(first, second);
```

**Review by Octol1ttle — APPROVED** [review]
now wait for @Apceniy

**Review by Octol1ttle — APPROVED** [review]

File: src/Commands/ToolsCommandGroup.cs [hunk 5]
```diff
@@ -266,25 +260,46 @@ private static void AppendBanInformation(MemberData memberData, StringBuilder bu
...
-                .WithColour(ColorsList.Red).Build();
+            description.Append(' ').Append(Messages.Default);
+        }
 
-            return await _feedback.SendContextualEmbedResultAsync(failedEmbed, ct);
+        description.AppendLine().Append("- ").Append(string.Format(
+            Messages.RandomMax, Markdown.InlineCode(max.ToString())));
+        if (secondNullable is null && first < 0)
+        {
+            description.Append(' ').Append(Messages.Default);
```
Comment (Octol1ttle) [inline src/Commands/ToolsCommandGroup.cs:289]: Put this into nested `if`-s

**Review by Octol1ttle — DISMISSED** [review]

File: src/Commands/ToolsCommandGroup.cs [hunk 6]
```diff
@@ -266,25 +260,46 @@ private static void AppendBanInformation(MemberData memberData, StringBuilder bu
...
-        return await SendRandomNumberAsync(max, min, user, currentUser, CancellationToken);
+        return await SendRandomNumberAsync(first, second, user, CancellationToken);
     }
 
-    private async Task<Result> SendRandomNumberAsync(int max, int min, IUser user, IUser currentUser, CancellationToken ct)
+    private async Task<Result> SendRandomNumberAsync(long first, long? secondNullable,
+        IUser user, CancellationToken ct)
     {
-        if (min > max)
+        var second = secondNullable ?? 0;
```
Comment (Apceniy) [inline src/Commands/ToolsCommandGroup.cs:269]:
```suggestion
        var secondDefault = 0;
        var second = secondNullable ?? secondDefault;
```

File: src/Commands/ToolsCommandGroup.cs [hunk 7]
```diff
@@ -266,25 +260,46 @@ private static void AppendBanInformation(MemberData memberData, StringBuilder bu
...
+        var min = Math.Min(first, second);
+        var max = Math.Max(first, second);
+
+        var i = Random.Shared.NextInt64(min, max + 1);
+
+        var description = new StringBuilder().Append("# ").Append(i);
+
+        description.AppendLine().Append("- ").Append(string.Format(
+            Messages.RandomMin, Markdown.InlineCode(min.ToString())));
+        if (secondNullable is null && first >= 0)
```
Comment (Apceniy) [inline src/Commands/ToolsCommandGroup.cs:280]:
```suggestion
        if (secondNullable is null && first >= secondDefault)
```

File: src/Commands/ToolsCommandGroup.cs [hunk 8]
```diff
@@ -266,25 +260,46 @@ private static void AppendBanInformation(MemberData memberData, StringBuilder bu
...
-            var failedEmbed = new EmbedBuilder().WithSmallTitle(
-                    Messages.RandomMinGreaterThanMax, currentUser)
-                .WithColour(ColorsList.Red).Build();
+            description.Append(' ').Append(Messages.Default);
+        }
 
-            return await _feedback.SendContextualEmbedResultAsync(failedEmbed, ct);
+        description.AppendLine().Append("- ").Append(string.Format(
+            Messages.RandomMax, Markdown.InlineCode(max.ToString())));
+        if (secondNullable is null && first < 0)
```
Comment (Apceniy) [inline src/Commands/ToolsCommandGroup.cs:287]:
```suggestion
        if (secondNullable is null && first < secondDefault)
```

**Review by Apceniy — CHANGES_REQUESTED** [review]
Replace magic 0s with a constant for the default value with meaningful name that reflects the purpose, improves flexibility for the possible future changes and reduces human errors

**Review by Octol1ttle — DISMISSED** [review]

File: src/Commands/ToolsCommandGroup.cs [hunk 9]
```diff
@@ -266,7 +266,8 @@ private static void AppendBanInformation(MemberData memberData, StringBuilder bu
     private async Task<Result> SendRandomNumberAsync(long first, long? secondNullable,
         IUser user, CancellationToken ct)
     {
-        var second = secondNullable ?? 0;
+        const int secondDefault = 0;
```
Comment (Apceniy) [inline src/Commands/ToolsCommandGroup.cs:269]: we're working with longs
```suggestion
        const long secondDefault = 0;
```

**Pull request merged**